### PR TITLE
update fetch calls to match template, submit renders mock data

### PR DESCRIPTION
**Pull request by aslink87**
now at /commodities there is no submit example button. instead the form submit fetches mock data no matter the field values. Until the api is functioning appropriately (distance values, landing pad sizes, etc need implemented), it is best we just return mock data.

I've updated the fetch calls at /commodities to match the template. I also expanded on the mock data to provide a more robust response

Added a swr hook for fetching mock data. It's time we start returning mock data as examples for all the pages we've made that have incomplete API endpoints.

### CODE REVIEW

File: app/commodities/page.client.tsx [hunk 1]
```diff
@@ -9,84 +9,65 @@ import {
...
-      setIsLoading(false);
-      setSubmitted(true);
-    });
-
-    if (response instanceof Error) {
-      setFetchError(response as Error);
-      return;
-    }
-    setCommodityResponse(response);
+    if (queryString !== '') await mutate();
```
Comment (aslink87) [inline app/commodities/page.client.tsx:54]: @nitishdhar I'd like your opinion here.

We call the swr hook with a queryString(URL). The problem is that the queryString will comprise of data from the form submission. Here I've commented out what that queryString will look like when fully implemented. So the question is what's the best way to pass the dynamic queryString to the swr hook?

Here I opted for setting the queryString as state when the form submits. It works fine but b/c of the async nature I added an if check to make sure the swr hook isn't called until the state is updated. If u remove the if check it still works but the network tab shows a 404 to the default state of the queryString. Is there a better way to pass a string comprised of form data to the swr hook? I guess I just think the if check is ugly but if it works, it works.

**Comment by aslink87** [comment]
Be advised that due to [error messages](https://github.com/ed-pilots-network/frontend/pull/126/checks?check_run_id=17067762451) I updated the playground route. #127

File: app/commodities/page.client.tsx (hunk 1 above)
Comment (nitishdhar) [reply app/commodities/page.client.tsx:54]: @aslink87 you are going right about it. Basically you would could use `useState` hooks for all individual form values as we would for normal forms and create the queryString with that -

```
  const [inputValue, setInputValue] = useState('');
  ....

  // Create a query string based on the input values
  const queryString = `/api/v1/${inputValue}.....`;
```

Now in the handleSubmit, you have two options, a) validate each input individually before calling mutate or use some general form validator that validates the entire form data at once using something like zod.

**Comment by nitishdhar** [comment]
> Be advised that due to [error messages](https://github.com/ed-pilots-network/frontend/pull/126/checks?check_run_id=17067762451) I updated the playground route. #127

Not sure why this was needed? Did you move these tests under template instead of just completely removed testing the error handling?

**Comment by aslink87** [comment]
> > Be advised that due to [error messages](https://github.com/ed-pilots-network/frontend/pull/126/checks?check_run_id=17067762451) I updated the playground route. #127
> 
> Not sure why this was needed? Did you move these tests under template instead of just completely removed testing the error handling?

Tbh I was in a time crunch here and opted to remove what was blocking while trying not to change your playground route too much. I thought you created the error pages in playground to show other devs what the error pages looked like. Looking at it now, they served also to provide us with our only means to test errors. When I have a moment, i'll make a commit here and put those back in. They may have to be more root level if that route is removed in prod. Thanks for looking this over.

**Review by nitishdhar — APPROVED** [review]

**Pull request merged**